### ISSUE TITLE
The API documentation now is called "sdk" in the top navigation bar

### ISSUE THREAD

**Issue by llucax** [frequenz-floss/frequenz-sdk-python#640]
### What happened?

The generated website now says "sdk" in the top navigation bar.

![image](https://github.com/frequenz-floss/frequenz-sdk-python/assets/1031485/09485d40-fdbb-4d13-9dd1-b7b693bc05e7)

### What did you expect instead?

It should say "API Reference", as it was up to v0.24.0

![image](https://github.com/frequenz-floss/frequenz-sdk-python/assets/1031485/ee17913e-255e-424c-b8f2-1fb49dc81a1d)

### Affected version(s)

v0.25.0

### Affected part(s)

Documentation (part:docs)

### Extra information

Nothing really changed, this is taken from https://github.com/frequenz-floss/frequenz-sdk-python/blame/v0.x.x/docs/SUMMARY.md, so this is probably a regression in a bumped dependency.

**Comment by llucax** [comment]
A bisection points to e7a3c3a1d5855f3de5a4072a338d236ff6a3cba8. So we'll have to have a look at what changed in repo config from 0.4.0 to 0.5.2.

**Comment by shsms** [comment]
But clicking on it takes us to this link directly: https://frequenz-floss.github.io/frequenz-sdk-python/v0.25/reference/frequenz/sdk/

and not to the top level page, like it used to.  That's very nice.  And the top level page for `API reference` appears to have gone away completely.

**Comment by llucax** [comment]
No, that was a separate issue, fixed in v0.6.0. We added the file `src/conftest.py` for testing the examples, and the documentation for it was generated too (not it is excluded), so the link would take you to `conftest` because it was the most top-level module.

**Comment by llucax** [comment]
The only change of dependencies I see there is in https://github.com/frequenz-floss/frequenz-repo-config-python/commit/d3cd805ac45f49d170efce5bb7b698933a16e671 (bumping from 9.1.16 to 9.1.21 of `mkdocs-material` and doing `pip install mkdocs-material==9.1.16` does solve the issue.

Strangely enough, doing a `pip install mkdocs-material==9.1.21` doesn't break it, but doing `pip install mkdocs-material==9.2.0` does, so I don't know what weird interaction was there but it seems like the update to 9.2.0+ is to blame ([release changelog](https://github.com/squidfunk/mkdocs-material/releases/tag/9.2.0)), which happened in 889f9a43, between v0.24.0 and v0.25.0.

**Comment by matthias-wende-frequenz** [comment]
removing the `section-index` plugin resolves the issue

**Comment by llucax** [comment]
Reported upstream: https://github.com/squidfunk/mkdocs-material/issues/5971

**Comment by llucax** [comment]
> removing the section-index plugin resolves the issue

Oh, interesting.

**Comment by llucax** [comment]
Removing `mkdocs-section-index` is not appropriate either, if we do so, we get:

![image](https://github.com/frequenz-floss/frequenz-sdk-python/assets/1031485/71f8e00c-239a-4eb8-a0ad-945ceb211a8a)

(Note that you need to click on the `index` menu for each package)

While with the plug-in we get no `index` on each package, which is what we want:

![image](https://github.com/frequenz-floss/frequenz-sdk-python/assets/1031485/56d36ed5-e50a-4e94-8f0e-c2cb0559783f)

**Comment by llucax** [comment]
`mkdocs-material` already closed the issue (as he is not looking at it saying it's an "upstream" problem), so trying with `mkdocs-section-index` and the most other likely cause of it:

* https://github.com/oprypin/mkdocs-section-index/issues/18

**Comment by llucax** [comment]
For now I think the easiest workaround is to downgrade `mkdocs-material`. Done in #642.

**Comment by llucax** [comment]
Marking as blocked as now we need to wait for upstream to see if we can upgrade the dependency again.

**Comment by llucax** [comment]
Unblocking as it is fixed upstream, but we are still switching to the native feature in `mkdocs-material`.
